File: RepoTransBench/target_projects/Python/Java/martin-majlis_Wikipedia-API/src/test/java/com/example/wikiapi/original/ExtractErrorsTest.java
package com.example.wikiapi.original;

import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.*;

class ExtractErrorsTest {
    WikipediaApi.Wikipedia wiki;

    @BeforeEach
    void setUp() {
        wiki = WikipediaApi.MockFactory.makeWikipedia();
    }

    @Test
    void testTitleBeforeFetching() {
        WikipediaApi.Page page = wiki.page("NonExisting");
        assertEquals("NonExisting", page.title);
    }

    @Test
    void testPageId() {
        WikipediaApi.Page page = wiki.page("NonExisting");
        assertEquals(-1, page.pageid);
    }
}

File: RepoTransBench/target_projects/Python/C++/martin-majlis_Wikipedia-API/tests/original/extract_errors_test.cpp
#include <gtest/gtest.h>
#include "wikipedia_api_mock.h"

class TestErrorsExtracts : public ::testing::Test {
protected:
    void SetUp() override {
        wiki = std::make_unique<Wikipedia>(mock_user_agent, "en");
        wiki->set_query_func(wikipedia_api_request(wiki.get()));
    }
    std::unique_ptr<Wikipedia> wiki;
};

TEST_F(TestErrorsExtracts, TitleBeforeFetching) {
    auto page = wiki->page("NonExisting");
    ASSERT_EQ(page.title(), "NonExisting");
}

TEST_F(TestErrorsExtracts, PageId) {
    auto page = wiki->page("NonExisting");
    ASSERT_EQ(page.pageid(), -1);
}